carbon-design-system/carbon-for-ibm-dotcom
# fix(structured-list & pricing-table): allow for rich text formatting

### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/8962

### Description
- This PR adds a `handleSlotChange` in the render function that takes the content of the `<dds-structured-list-cell>`  and `<dds-pricing-table-header-cell-description>` components and appends it to the shadow root for flexibility styling using a class name.
- This allows adopters like AEM to add HTML tags that are used within the WYSIWYG (eg. `<p>`,`<ul>`, `<ol>`) and ones that are defined in the design specs.

### Changelog

**New**

- Handles `mutationObserver` event within the `<dds-structured-list-cell>`  and `<dds-pricing-table-header-cell-description>` components.

**Changed**

- `<dds-pricing-table-cell-annotation>` now extends from `<dds-pricing-table-header-cell-description>`
- Update stories to show examples of rich text format vs custom elements from Carbon Core's `<<bx-unordered-list>`

![Screen Shot 2022-06-13 at 4 32 43 PM](https://user-images.githubusercontent.com/1815714/173441754-377954dc-3854-46cb-a378-f021a0dcb567.png)
![Screen Shot 2022-06-13 at 4 32 01 PM](https://user-images.githubusercontent.com/1815714/173441771-8c7007c7-f1f0-4f02-8c37-f46464e3fb55.png)

### Testing Instructure
- Visit the "Structure List" component - `?path=/story/components-structured-list--default`
- Ensure that the `<p>` and `<ul>` tags are rendering correctly and follows the design specs.
- Visit the "Pricing Table" component - `?path=/story/components-pricing-table--default`
- Ensure that the `<p>` and `<ul>` tags are rendering correctly and follows the design specs.
